core/pkg/server/writer.go

```diff
@@ -135,27 +135,36 @@ func (w *Writer) Close() {
 
 // writeRecord Writing messages to the append-only log,
 // and passing them to the sender.
-// We ensure that the messages are written to the log
-// before they are sent to the server.
+// Ensure that the messages are numbered and written to the transaction log
+// before network operations could block processing of the record.
 func (w *Writer) writeRecord(record *service.Record) {
 	switch record.RecordType.(type) {
 	case *service.Record_Request:
 		w.fwdRecord(record)
 	case nil:
 		w.logger.Error("writer: writeRecord: nil record type")
 	default:
-		w.fwdRecord(record)
+		// applyRecordNumber() should be called before passing the record to another goroutine
```

Comment:
@kptkin am i going crazy? i was sure we swapped the order, no??

Reply:
i think we didn't merge it

Reply:
dang